### ISSUE TITLE
Local PubSub Helper

### ISSUE THREAD

**Issue by lucasoares** [googleapis/google-cloud-java#6295]
I would like a class like the [LocalDatastoreHelper ](https://github.com/lucasoares/google-cloud-java/blob/master/google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/testing/LocalDatastoreHelper.java) to start the PubSub Emulator directly from my code.

I want to contribute on this if you guys plan to continue supporting this kind of feature.

Thanks.

**Comment by elefeint** [comment]
There is a [`PubSubEmulator`](https://github.com/spring-cloud/spring-cloud-gcp/blob/44d394e183d2b3f56985d51e2fe86eb136f5ccfd/spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderEmulatorTests.java#L45) JUnit rule ([example use)](https://github.com/spring-cloud/spring-cloud-gcp/blob/44d394e183d2b3f56985d51e2fe86eb136f5ccfd/spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderEmulatorTests.java#L45) in spring-cloud-gcp that can be either ported or used as inspiration.

**Comment by lucasoares** [comment]
Yes, it's kinda like the LocalDatastoreHelper and it is just the execution of the beta emulator command.

Thanks for the inspiration, maybe I will use it on my projects. I asked here before developing anything to be sure this are a feature that will be supported to other GCP resources. I saw only the Datastore one.
